simplerunner.lua

```diff
@@ -63,53 +109,276 @@ function runbench(name, count)
+    print(string.format("  Mean: %f +/- %f, min %f, max %f", stats.mean, stats.cinterval, stats.min, stats.max))
+    if jitstats then
+        jitstats.print()
+    end
+end
+
+function calculate_stats(times, start)
+    local stats = {}
+    start = start or 1
+    local count = #times - start-1
+    
+    local min, max, total = 10000, 0, 0
+    for i = start, #times do
+        local time = times[i]
+        total = total + time
+        min = math.min(time, min)
+        max = math.max(time, max)
+    end
+
+    stats.min = min
+    stats.max = max  
+    stats.mean = total / count
+    
+    local variance = 0
+    for i = start, count do
+        local diff = times[i] - stats.mean
+        variance = variance + diff*diff
+    end
+    
+    stats.stddev = math.sqrt(variance / count)
+    -- 95% confidence interval 
+    stats.cinterval = 1.960 * stats.stddev/math.sqrt(count)   
+    return stats
+end
+
+-- Command line parsing system borrowed heavily from dynasm.lua in LuaJIT
+local opt_map = {}
+local g_opt = {}
+
+-- Print error and exit with error status.
+local function opterror(...)
+    io.stderr:write("simplerunner.lua: ERROR: ", ...)
+    io.stderr:write("\n")
+    os.exit(1)
+end
+
+-- Get option parameter.
+local function optparam(args, argtype)
+    local argn = args.argn
+    local p = args[argn]
+    if not p then
+        opterror("missing parameter for option `", opt_current, "'.")
+    end
+    args.argn = argn + 1
+    
+    if argtype == "number" then
+        local value = tonumber(p)
+        if not value then
+            opterror("expected number for parameter option `", opt_current, "'.")
+        end
+        p = value
+    end
+    
+    return p
+end
+
+-- Short aliases for long options.
+local opt_alias = {
+    h = "help", ["?"] = "help", V = "version",
+    b = "bench", c = "count", s = "scaling",
+}
+
+-- Print help text.
+function opt_map.help()
+  io.stdout:write[[
+
+Usage: simplerunner [OPTION]... benchmark/count  c|-
+
+  -h, --help           Display this help text.
+  -V, --version        Display version.
+
+  -b, --bench name     Name of benchmark to run.
+  -c, --count num      Number of times to run the benchmarks.
+  -s, --scaling num    Number of interations to run the benchmarks.
+
+  --jitstats           Collect and print jit statisitcs from running the benchmarks.
+  --benchloadjitstats  Also collect jitstats when loading a benchmark 
+]]
+  exit(0)
+end
+
+-- Misc. options.
+function opt_map.count(args) g_opt.count = optparam(args, "number") end
+function opt_map.bench(args) g_opt.bench = optparam(args) end
+function opt_map.scaling(args) g_opt.scaling = optparam(args, "number") end
+function opt_map.jitstats() g_opt.jitstats = true end
+function opt_map.benchloadjitstats() loading_jitstats = true end
+function opt_map.nogc() g_opt.nogc = true end
+function opt_map.jitlog() g_opt.jitlog = true end
+
+------------------------------------------------------------------------------

```

Comment:
Not sure this line is helpful.

Reply:
its was there from the code i copied it from although the ordering is slightly different

Reply:
Copied from where? What's the license of the thing that was copied?